src/adapter/src/coord/appends.rs

```diff
@@ -208,17 +224,40 @@ impl<S: Append + 'static> Coordinator<S> {
                     .send(Message::GroupCommit)
                     .expect("sending to internal_cmd_tx cannot fail");
             });
-        } else if self
+        } else {
+            self.group_commit().await;
+        }
+    }
+
+    /// Commits all pending write transactions at the same timestamp. All pending writes will be
```

Comment:
This comment says "commits all pending write transactions", but that's not quite true. It will only do that if it can acquire a lock. But it will always do the system table updates even without a lock. I think it's worth describing this guarantee and behavior, because that's explicitly depended on by `send_builtin_table_updates` to function correctly.

Reply:
Good catch, just updated the comment

Reply:
and some other ones around it.